VimeoUpload/Upload/Descriptor System/New Upload (Private)/UploadDescriptor.swift

```diff
@@ -189,6 +189,8 @@ open class UploadDescriptor: ProgressDescriptor, VideoDescriptor
         let path = URL.uploadDirectory().appendingPathComponent(fileName).appendingPathExtension(fileExtension).absoluteString
         
         self.url = URL(fileURLWithPath: path)
+        
+        self.uploadStrategy = aDecoder.decodeObject(of: UploadStrategy.self, forKey: type(of: self).UploadStrategyKey)
```

Comment:
Will this give us any problems with `NSSecureCoding` if we're archiving a subclass of `UploadStrategy`? Should we be using `StreamingUploadStrategy` here instead?

Reply:
Wondering if we should give thought to the fact that we're providing an abstract class folks will need to subclass and then if they make a custom one, they'll need to handle the decoding with their concrete class. Unless we think providing a default strategy is sufficient.